### PR TITLE
Add if, else if and else to javascript

### PR DESCRIPTION
**Pull request by TheOtterlord**
Adds the `if`, `else if` and `else` JavaScript implementation

### CODE REVIEW

File: src/compiler/target/javascript/mod.rs [hunk 1]
```diff
@@ -68,9 +68,50 @@ impl JavaScript {
...
+    let statement = *if_.0;
+    let mut prefix = Inline::from_str("if (");
+    self.action(statement, &mut prefix, true);
+    prefix.code(")");
+    let mut actions = Block::new();
+    for action in if_.1.list {
+      self.action(action, &mut actions, false);
+    }
+    lb.function(prefix, actions);
+    // else if
```
Comment (mjarkk) [inline src/compiler/target/javascript/mod.rs:91]:
```suggestion

    // else if
```

File: src/compiler/target/javascript/mod.rs [hunk 2]
```diff
@@ -68,9 +68,50 @@ impl JavaScript {
...
+    let statement = else_if[0].0.clone();
+    let mut prefix = Inline::from_str("else if (");
+    self.action(statement, &mut prefix, true);
+    prefix.code(")");
+    let mut actions = Block::new();
+    for action in else_if[0].1.clone().list {
+      self.action(action, &mut actions, false);
+    }
+    lb.function(prefix, actions);
+    // else
```
Comment (mjarkk) [inline src/compiler/target/javascript/mod.rs:101]:
```suggestion

    // else
```

File: src/compiler/target/javascript/mod.rs [hunk 3]
```diff
@@ -68,9 +68,50 @@ impl JavaScript {
...
+    // if
+    let statement = *if_.0;
+    let mut prefix = Inline::from_str("if (");
+    self.action(statement, &mut prefix, true);
+    prefix.code(")");
+    let mut actions = Block::new();
+    for action in if_.1.list {
+      self.action(action, &mut actions, false);
+    }
+    lb.function(prefix, actions);
```
Comment (mjarkk) [inline src/compiler/target/javascript/mod.rs:90]: We repeat this code multiple times here, might be worth it to move it to a function to reduce the amount of code.

Something like this:  
```rust
fn if_block(lb: &mut impl BuildItems, body: Actions, start_prefix_with: &'static str, add_to_prefix: impl FnOnce(prefix: Inline)) {
    let statement = *if_.0;
    let mut prefix = Inline::from_str(start_prefix_with);
    add_to_prefix(prefix);

    let mut actions = Block::new();
    for action in body {
      self.action(action, &mut actions, false);
    }
    lb.function(prefix, actions);
}
```

The function can be called using:
```rust
let statement = *if_.0;
if_block(lb, if_.1.list, "if (", move |prefix| {
    self.action(statement, &mut prefix, true);
    prefix.code(")");
})
```

Comment (TheOtterlord) [reply src/compiler/target/javascript/mod.rs:90]: This is appearing as invalid (line `1`). I've never seen inline implementations and not sure how they work. 

```rust
impl FnOnce(prefix: Inline)
```

Comment (mjarkk) [reply src/compiler/target/javascript/mod.rs:90]: These inline functions can be a bit tricky, if it doesn't work you might want to try some of these:  
- not use the move keyword
- add the `Inline` type to the inline function
- the body argument to a reference
- change the body argument to a implementation of iterator, and add .iter() to the if_1.list

Comment (TheOtterlord) [reply src/compiler/target/javascript/mod.rs]: What do you think of this? 07178ae28aad2c2acf3107f274071d7f9bf6f038

Comment (mjarkk) [reply src/compiler/target/javascript/mod.rs]: I'll checkout this branch and see if i can implement it :)

Comment (TheOtterlord) [reply src/compiler/target/javascript/mod.rs]: Looks much better. Thanks :-)

**Comment by mjarkk** [comment]
I have made some changes to add what i suggested.  
i also changed the if else into a for loop because if we can't assume all for loops have an else if statement and a user could use multiple else ifs.

**Comment by TheOtterlord** [comment]
Yeah, it looks great. Thanks for the help. (Forgot about multiple if/else. Getting to use to match 🤣 )

**Comment by mjarkk** [comment]
Aii, i'll merge this in :)

**Pull request merged**